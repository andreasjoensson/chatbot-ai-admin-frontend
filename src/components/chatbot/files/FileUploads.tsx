import React, { ReactElement } from 'react';

interface FileUploadsProps {
  getRootProps: () => Record<string, any>;
  getInputProps: () => Record<string, any>;
  selectedFiles: File[];
  handleSaveFile: () => void;
  url: string;
  scrapeAllUrls: () => void;
  setUrl: (value: string) => void;
  handleUrlSubmit: () => void;
}

const FileUploads: React.FC<FileUploadsProps> = ({
  getRootProps,
  getInputProps,
  selectedFiles,
  handleSaveFile,
  url,
  setUrl,
  scrapeAllUrls,
  handleUrlSubmit,
}: FileUploadsProps): ReactElement => {
  return (
    <div>
      <div className="mb-4">
        <h2 className="text-lg font-semibold">File Upload</h2>
        <div
          {...getRootProps()}
          className="border-dashed border-2 border-gray-300 p-4"
        >
          <input {...getInputProps()} />
          <p>Drag 'n' drop some files here, or click to select files</p>
          {selectedFiles.map((file: File) => (
            <p key={file.name}>{file.name}</p>
          ))}
        </div>
      </div>
      <div className="mb-4">
        {selectedFiles.length > 0 && (
          <button
            onClick={handleSaveFile}
            className="bg-black text-white p-2 mt-2 rounded-md hover:bg-blue-600"
          >
            Save File
          </button>
        )}
      </div>
      <div className="mb-4">
        <h2 className="text-lg font-semibold">URL</h2>
        <input
          type="text"
          placeholder="Enter a base URL"
          className="border border-gray-300 p-2 w-full mb-2"
          value={url}
          onChange={(e) => setUrl(e.target.value)}
        />
        <button
          onClick={scrapeAllUrls}
          className="bg-black text-white p-2 mt-2 rounded-md hover-bg-blue-600"
        >
          Scrape all URLs
        </button>
      </div>
      <div className="mb-4">
        <h2 className="text-lg font-semibold">URL</h2>
        <input
          type="text"
          placeholder="Enter a URL"
          className="border border-gray-300 p-2 w-full mb-2"
          value={url}
          onChange={(e) => setUrl(e.target.value)}
        />
        <button
          onClick={handleUrlSubmit}
          className="bg-black text-white p-2 mt-2 rounded-md hover-bg-blue-600"
        >
          Submit URL
        </button>
      </div>
    </div>
  );
};

export default FileUploads;
