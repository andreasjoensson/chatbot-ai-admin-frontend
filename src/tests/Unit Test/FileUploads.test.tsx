import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import FileUploads from '../../components/chatbot/files/FileUploads';

describe('FileUploads', () => {
  const defaultProps = {
    getRootProps: jest.fn(),
    getInputProps: jest.fn(),
    selectedFiles: [],
    handleSaveFile: jest.fn(),
    url: '',
    setUrl: jest.fn(),
    handleUrlSubmit: jest.fn(),
  };

  it('should render without errors', () => {
    const { container } = render(<FileUploads {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should display a file upload area', () => {
    const { getByText } = render(<FileUploads {...defaultProps} />);
    expect(getByText(/Drag 'n' drop some files here/i)).toBeInTheDocument();
  });

  it('should display selected file names', () => {
    const selectedFiles = [new File(['file content'], 'file.txt')];
    const { getByText } = render(
      <FileUploads {...defaultProps} selectedFiles={selectedFiles} />,
    );
    expect(getByText('file.txt')).toBeInTheDocument();
  });

  it('should call handleSaveFile when "Save File" button is clicked', () => {
    const { getByText } = render(
      <FileUploads
        {...defaultProps}
        selectedFiles={[new File(['file content'], 'file.txt')]}
      />,
    );
    const saveButton = getByText('Save File');
    fireEvent.click(saveButton);
    expect(defaultProps.handleSaveFile).toHaveBeenCalled();
  });

  it('should display an input for entering a URL', () => {
    const { getByPlaceholderText } = render(<FileUploads {...defaultProps} />);
    expect(getByPlaceholderText('Enter a URL')).toBeInTheDocument();
  });

  it('should call setUrl when URL input value is changed', () => {
    const { getByPlaceholderText } = render(<FileUploads {...defaultProps} />);
    const urlInput = getByPlaceholderText('Enter a URL');
    fireEvent.change(urlInput, { target: { value: 'https://example.com' } });
    expect(defaultProps.setUrl).toHaveBeenCalledWith('https://example.com');
  });

  it('should call handleUrlSubmit when "Submit URL" button is clicked', () => {
    const { getByText } = render(<FileUploads {...defaultProps} />);
    const submitButton = getByText('Submit URL');
    fireEvent.click(submitButton);
    expect(defaultProps.handleUrlSubmit).toHaveBeenCalled();
  });
});
