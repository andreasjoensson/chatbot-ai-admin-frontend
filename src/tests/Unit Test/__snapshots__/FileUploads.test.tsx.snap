// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploads should render without errors 1`] = `
<div>
  <div>
    <div
      class="mb-4"
    >
      <h2
        class="text-lg font-semibold"
      >
        File Upload
      </h2>
      <div
        class="border-dashed border-2 border-gray-300 p-4"
      >
        <input />
        <p>
          Drag 'n' drop some files here, or click to select files
        </p>
      </div>
    </div>
    <div
      class="mb-4"
    />
    <div
      class="mb-4"
    >
      <h2
        class="text-lg font-semibold"
      >
        URL
      </h2>
      <input
        class="border border-gray-300 p-2 w-full mb-2"
        placeholder="Enter a URL"
        type="text"
        value=""
      />
      <button
        class="bg-black text-white p-2 mt-2 rounded-md hover-bg-blue-600"
      >
        Submit URL
      </button>
    </div>
  </div>
</div>
`;
